src/test/regress/specs/isolation_progress_monitoring.spec

```diff
@@ -19,6 +19,22 @@ setup
 	RETURNS TABLE(step int, progress bigint)
 	AS 'citus'
 	LANGUAGE C STRICT;
+
+	CREATE FUNCTION sample_operation(command_type bigint, lockid bigint, progress bigint)
+		RETURNS VOID AS $$
+	BEGIN
+		PERFORM create_progress(command_type, 2);
+		PERFORM pg_advisory_lock(lockid);
+		PERFORM update_progress(0, progress);
+		PERFORM pg_advisory_lock(lockid+1);
+		PERFORM update_progress(1, progress);
+		PERFORM pg_advisory_lock(lockid+2);
+		PERFORM finish_progress();
+		PERFORM pg_advisory_unlock(lockid);
+		PERFORM pg_advisory_unlock(lockid+1);
+		PERFORM pg_advisory_unlock(lockid+2);
```

Comment:
These locks are making me a little dizzy :). Comments on what's going on would be helpful.

You could use `pg_advisory_xact_lock` here to avoid having to unlock.